daml-assistant/src/DA/Daml/Assistant/Install/Path.hs

```diff
@@ -76,13 +79,49 @@ regQueryStringValue key valueName =
             failUnlessSuccess "regQueryDefaultValue" $
                 c_RegQueryValueEx p_key c_valueName nullPtr nullPtr c_value p_valueLen
             peekTString (castPtr c_value)
-#else
-updatePath _ output targetPath = suggestAddToPath output targetPath
-#endif
 
 suggestAddToPath :: (String -> IO ()) -> FilePath -> IO ()
 suggestAddToPath output targetPath = do
     -- Ask user to add .daml/bin to PATH if it is absent.
     searchPaths <- map dropTrailingPathSeparator <$> getSearchPath
     when (targetPath `notElem` searchPaths) $ do
         output ("Please add " <> targetPath <> " to your PATH.")
+#else
+updatePath _ output targetPath = do
+  searchPaths <- map dropTrailingPathSeparator <$> getSearchPath
+  unless (targetPath `elem` searchPaths) $ do
+    shellPath <- getEnv "SHELL"
+    let shell = takeFileName shellPath
+    case shellConfig shell targetPath of
+      Nothing -> output setPathManualMsg
+      Just (configFile, cmd) -> do
+        home <- getHomeDirectory
+        let cfgFile = home </> configFile
+        cfgFileExists <- doesFileExist cfgFile
+        if cfgFileExists
+          then do
+            content <- readFile cfgFile
+            unless (cmd `elem` lines content) $ do
+              answer <- prompt "Add DAML Assistant executable to your PATH?" "Yes" ["No"]
+              when (answer `elem` ["Yes", "yes", "y", "Y"]) $ appendFile cfgFile cmd
+          else output setPathManualMsg
+  where
+    setPathManualMsg = "Please add " <> targetPath <> " to your PATH."
+#endif
+
+prompt :: String -> String -> [String] -> IO String
+prompt msg def others = do
+  putStrLn $ msg <> (unwords $ " [" <> def <> "]" : others)
+  ans <- getLine
+  return $
+    if null ans
+      then def
+      else ans
+
+shellConfig :: String -> FilePath -> Maybe (FilePath, String)
+shellConfig shell targetPath =
+  case shell of
+    "zsh" -> Just (".zshrc", "export PATH=$PATH:" <> targetPath)
+    "bash" -> Just (".bashrc", "export PATH=$PATH:" <> targetPath)
```

Comment:
`.bashrc`, not `.bash_profile`? Likewise, `.zshrc` instead of `.zprofile`? I think, on a Mac (at least), the latter two are preferred for PATH changes, and I don't have a `.bashrc` or `.zshrc` but I hove the other two.

Reply:
hmm, on my linux box it's .zshrc and .bashrc and I don't have the `profile` ones.

Reply:
Yeah, all my paths are configured in `.zprofile` as well.

Reply:
I think `.bash_profile` and `.zprofile` are more sensible defaults on Mac at least. (I think using `isMac` from `System.Info.Extra` is best to distinguish the cases here. http://hackage.haskell.org/package/extra-1.7.4/docs/System-Info-Extra.html#v:isMac)

Reply:
so apparantly the difference is that `.profile` and `.bash_profile` are read for login shells, while `.bashrc` is read for interactive shells. The recommendation is to set PATH in `.profile`, so it's available also for non-interactive shells after login. If we put it in `.profile/.zprofile/.bash_profile` the PATH will however only be set after logout/login again.

Reply:
Except on macos which does read `.profile` even for non-login shells (or pretends that everything is a login shell I forgot). Our instructions so far deliberately explained `.*profile*` and rebooting on Linux iirc so I would stick to that.

Reply:
I agree with @cocreature here.

Reply:
Then we should also inform the user that one has to logout/login again. We could also set the PATH in the current terminal, so that at least it's available there right after installation.

Reply:
Most of our tutorials open multiple terminals at some point so I think changing the current terminal is just going to result in confusion.

Reply:
Warning the user that they either need to logout/login seems like a good idea. Even if restarting the terminal is enough on MacOS, we should probably still tell them the same to avoid the effects @cocreature described. Maybe it's a good idea to tell people that they _might_ need to logout/login to pick up the `PATH` variable to not annoy people who know they don't need to.

Reply:
Well, you could just tell people to Quit (Cmd-Q) the Terminal on macOS. This kills all terminal windows and ensures that the next time you open the Terminal.

Alternatively, and I think this might be the right approach, we should include a script in our DAML installation (maybe at `~/.daml/setup.sh`) that people can `source` to get the `PATH` variable right away, and we give this as an option to the user when they finish the installation. This seems to be the standard practice for install tools that modify PATH variables.

Reply:
ok, we set '.profile' now for 'zsh', 'bash' and 'sh'. We don't do anything if the file doesn't exist.